src/vuefire.js

```diff
@@ -61,11 +85,7 @@ function bind (vm, key, source) {
   if (!isObject(source)) {
     throw new Error('VueFire: invalid Firebase binding source.')
   }
-  // get the original ref for possible queries
-  var ref = source
-  if (typeof source.ref === 'function') {
```

Comment:
I think we still need something like this since `source` might be a Firebase query, not a Firebase reference. So later on when we call methods off it thinking it's a reference, things would fail. The check does need to be updated to support `2.x.x` and `3.x.x` though:

```
if (typeof source.ref === 'function') {
  ref = source.ref()
} else (typeof source.ref === 'object') {
  ref = source.ref
}
```